src/library_syscall.js

```diff
@@ -93,9 +93,15 @@ var SyscallsLibrary = {
     doReadlink: function(path, buf, bufsize) {
       if (bufsize <= 0) return -ERRNO_CODES.EINVAL;
       var ret = FS.readlink(path);
-      ret = ret.slice(0, Math.max(0, bufsize));
-      writeStringToMemory(ret, buf, true);
-      return ret.length;
+
```

Comment:
This is the only location in the codebase where we explicitly want to not null-terminate. I have no idea why this function is specced like it is, but since this is a single occurrence, it is good to just special case this pattern here, where we use `stringToUTF8` to write the string, and then restore the character that was underneath the null byte that was written.